src/main/java/org/reactome/addlinks/referencecreators/EnsemblReferenceCreator.java

```diff
@@ -0,0 +1,115 @@
+import org.gk.persistence.MySQLAdaptor;
+
+import java.util.*;
+
+public class EnsemblReferenceCreator extends SimpleReferenceCreator<Map<String, List<String>>>{
+
+    public EnsemblReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB)
+    {
+        super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB);
+    }
+
+    public EnsemblReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB, String refCreatorName)
+    {
+        super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB, refCreatorName);
+    }
+
+    //TODO: Add unit tests
+
+    /**
+     * Find all Ensembl identifiers connected to an RGP instance in the Reactome DB and create a DatabaseIdentifier cross-reference instance for them.
+     * @param personID - Newly created Identifiers will be associated with an InstanceEdit. That InstanceEdit will be associated with the Person entity whose ID == personID
+     * @param mappings - Large mapping structure generated in FileProcessor step.
+     * @param sourceReferences - A list of GKInstance objects that were in the database. References will be created if they are in the keys of mapping and also in sourceReferences.
+     * This is to handle cases where a mapping comes from a third-party file that may contain source identifiers that are in *their* system but aren't actually in Reactome.
+     * @throws Exception
+     */
+    @Override
+    public void createIdentifiers(long personID, Map<String, Map<String, List<String>>> mappings, List<GKInstance> sourceReferences) throws Exception {
+
+        // Iterate through each ReferenceGeneProduct (RGP) instance in the DB.
+        logger.info("Creating Ensembl cross-references");
+        for (GKInstance sourceInst : sourceReferences) {
+            String sourceIdentifier = (String) sourceInst.getAttributeValue(ReactomeJavaConstants.identifier);
+            // Determine which species the instance pertains too. Retrieve corresponding mapping structures.
+            GKInstance speciesInst = (GKInstance) sourceInst.getAttributeValue(ReactomeJavaConstants.species);
+            String biomartSpeciesName = getBiomartSpeciesName(speciesInst.getDisplayName());
+            // Iterate through each Ensembl Protein identifier. Retrieve any corresponding Transcript and Gene
+            // identifiers and add each to a set. These will be used to create cross references to Ensembl from Reactome.
+            Set<String> ensemblIds = collectEnsemblIdentifiers(mappings, sourceIdentifier, biomartSpeciesName);
+            // Iterate through each Ensembl identifier and create a ? instance.
+            for (String ensemblId : ensemblIds) {
+                if (!this.checkXRefExists(sourceInst, ensemblId))
+                {
+                    if (!this.testMode)
```

Comment:
Since the only executable statement is the `createIdentifier` statement, these two conditions could be combined to:
```
if (!this.checkXRefExists(sourceInst, ensemblId) && !this.testMode)
{
```